test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -1797,34 +1779,41 @@ public void TempData_CanSetAndGetValues()
             Assert.Equal(input, result);
         }
 
-        private static Controller GetController(IModelBinder binder, IValueProvider provider)
+        private static Controller GetController(IModelBinder binder, IValueProvider valueProvider)
         {
             var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
             var httpContext = new DefaultHttpContext();
-            var actionContext = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
 
             var viewData = new ViewDataDictionary(metadataProvider, new ModelStateDictionary());
             var tempData = new TempDataDictionary(Mock.Of<IHttpContextAccessor>(), Mock.Of<ITempDataProvider>());
 
-            var bindingContext = new ActionBindingContext()
+            var controllerContext = new ControllerContext()
             {
-                ModelBinder = binder,
-                ValueProvider = provider,
+                HttpContext = httpContext,
                 InputFormatters = new List<IInputFormatter>(),
-                ValidatorProvider = new DataAnnotationsModelValidatorProvider(
-                    options: null,
-                    stringLocalizerFactory: null)
+                ModelBinders = new List<IModelBinder>()
```

Comment:
Yeah I was talking about these specific code snippets where there's only one type in the array, and due to `IList<T>` covariance and array implementing `IList<T>` that it might just work. No biggie.